support/src/main/java/com/shopify/graphql/support/Input.java

```diff
@@ -0,0 +1,26 @@
+package com.shopify.graphql.support;
+
+import java.io.Serializable;
+
+/**
+ * Created by henrytao on 9/7/17.
+ */
+
+public final class Input<T> implements Serializable {
+
+  public final T value;
```

Comment:
I think I would prefer to use a getter here instead of a public field. Give us more flexibility in the future.

Also, should this be final? Do you have to construct a new input object if you want to change the wrapped value?

Reply:
Yes it was an intention to make it immutable, following by best practises and similar to java Optional, guava Optional data types.

Reply:
And as for the getter  👍